### DISCUSSION TITLE
baseURL is not being used when deploying to github

### DISCUSSION THREAD

**Discussion by 1000camels** [thegetty/quire#659] [Get Help]
My config/environments/github.yml has two lines:

baseURL: "https://githubusername.github.io/repo/" # I have the correct address in mine
canonifyURLs: true

My images are working and the splash image on the cover is working, but all other hero images in page and essay md files does not render. The urls are neither relative nor include my baseURL (ie. style="background-image: url('/img/figures/image1.jpg');"

What am I doing wrong? I am still using Quire (v0)

**Comment by geealbers** [comment]
@1000camels, three ideas to try:

1. It might be the same as #444 -- the solution there was to delete the `config/_default/config.yml` if it exists in your project and then redeploy
2. If that doesn't work, you can go into `themes/default/layouts/partials/page-header.html` and `themes/default/layouts/splash/single.html` and change `relURL` to `absURL` (it occurs once in each of those files) and try to redeploy
3. Finally, if those don't work, there's a step for GitHub deploy where you need to make a change to `themes/default/webpack/webpack.config.prod.js`. It's step #7 here: https://quire.getty.edu/docs-v0/site-deploy/#github-pages. Make that change if you haven't already and redeploy

**Reply by 1000camels** [reply]
I had already done #1, and #3 was already set.
I copied the layouts and made the changes.
Unfortunately that did not work. I am not sure seeing my github baseURL in background-image only /

**Reply by geealbers** [reply]
Hm. I'm not sure what's going on. Could I get access to the repo?

**Comment by 1000camels** [comment]
You should have access now-invite via email

**Reply by geealbers** [reply]
Got it, thanks! When I test it locally, if I run `quire site --env=github` off the `main` branch, I _am_ getting the baseURL on those hero images as expected, like this one from `content/symposium/essay-fister.md`:

```html
<section class=" quire-page__header hero__image" style="background-image: url('https://denverartmuseum.github.io/her-brush/img/figures/2018_155.jpg');">
```

And `quire site --env=github` is the command used if you're running the `bin/deploy.sh` script. Can you try it once more? Or let me know how you're deploying if it's different, or if you're working from a branch other than `main`?

**Comment by 1000camels** [comment]
I will try later but I am just running bin/deploy.sh

**Comment by 1000camels** [comment]
Alright I tried just the quire site --env=github command and my background-image url's are still /img/ instead of the full domain.
I also checked out a new version of the repo and tried it, and it was the same. So it something in my environment.

My version of quire is v0.20.4.
I can upgrade, but I am not ready to go to v1 yet

I am also running this on MacOS. Happy to provide any other details

**Reply by geealbers** [reply]
I'm on a mac and running Quire v0.20.4 as well, and Node v14.18.1. You shouldn't upgrade to Quire v1 as it's a completely different architecture and won't be able to run your v0 project at all.

One other thought, is it possible you have an older version of Hugo installed than what Quire v0.20.4 is using? if you run `hugo version` it should output either `v0.82.1` (which is the version Quire v0.20.4 is pegged to), or `command not found` (which just means that you don't have a local version installed outside of the quire-cli package). Either of these two responses would be fine, anything else and that might be the source of the problem, especially if it's a much earlier version.

**Comment by 1000camels** [comment]
Discovered that I am running Hugo 0.92.2, installed via homebrew.
I managed to uninstall it. hugo not longer worked from the command line,
but quire site still worked.
I chose to reinstall ***@***.***

Still no luck.

I did install the newer Quire. I must have done it locally as it it is not
what appears via quire -V
But perhaps this is what messed up my environment.

What is the best way to run both or to at least clear v1, so I can keep
working on v0, which is more important right now?

Regards,
Darcy

**Reply by geealbers** [reply]
@1000camels because you answered through email, GitHub garbled part of your message. I'm seeing "I chose to reinstall ***@***.***" which I think was meant to maybe be a specific version of Quire? but Gtihub thought it was an email address and so anonymized it. What was it that you reinstalled?

And for running multiple versions, we use [nvm (node version manager)](https://github.com/nvm-sh/nvm). It allows you to have multiple versions of node installed and easily switch back and forth between them with a command. And so with nvm, I have node 14.18.1 installed and with that Quire 0.20.4, and then I have node 18.12.1 installed and with that Quire 1.0.0-rc-0. Because Quire is installed with a specific version of node that's been installed with nvm, when I switch node versions, Quire versions are automatically switched too.

**Comment by 1000camels** [comment]
If you run the following, do you see the correct github.io url? I do not,
which suggests to me that something is not reading the configs correctly

quire debug --env=github | grep baseURL

**Reply by geealbers** [reply]
When running `quire debug --env=github | grep baseURL` I don't get the correct URL either. It looks like it's drawing everything from the main `config.yml` file. This might be more an issue with `quire debug` though. I think maybe it's just not getting the `--env` flag passed in. But honestly, i don't think I've ever even used `quire debug` myself! so I'm not sure

**Comment by 1000camels** [comment]
I am really at a loss as what to do.

**Reply by geealbers** [reply]
Likewise! Good idea to have Matt test it on his end. Do you also want to have a call where we can share screens and walk through it together? Maybe that will help us figure out why we're not getting the same results. Drop us an email (quire@getty.edu) to set up a time. Tuesday 2/14 might be our best shot as I'll be traveling Wednesday, at the [College Art Association conference](https://www.collegeart.org/programs/conference/conference2023) Thurs–Sat, and then not back in the office until Tues 2/21.

**Reply by Erin-Cecele** [reply]
@1000camels I know many months have passed but I wanted to touch base and see if you were eventually able to resolve this issue. Thank you!
